test/archive/har.d

```diff
@@ -0,0 +1,198 @@
+                return No.success;
+            }
+        }
+        else
+        {
+            if (verbose)
+                infoFile.writefln("mkdir %s", dir.formatDir);
+            mkdirRecurse(dir);
+        }
+        return Yes.success;
+    }
+
+    Flag!"success" extract()
+    {
+        if (outputDir is null)
+        {
+            outputDir = harFilename[0 .. $ - ".har".length];
+            if (verbose)
+                infoFile.writefln("Using default output directory %s", outputDir.formatDir);
+        }
+    
+        if (!exists(harFilename))
+        {
+            infoFile.writefln("Error: %s does not exist", harFilename.formatFile);
+            return No.success;
+        }
+        auto harFile = File(harFilename, "r");
+    
+        if (exists(outputDir))
+        {
+            if (!isDir(outputDir))
+            {
+                infoFile.writefln("Error: output directory %s already exists and is not a directory", outputDir.formatDir);
+                return No.success;
+            }
+            // Maybe a warning if directory is not empty or something
+            if (verbose)
+                infoFile.writefln("output directory %s already exists", outputDir.formatDir);
+        }
+        else
+        {
+            if (verbose)
+                infoFile.writefln("mkdir %s", outputDir.formatDir);
+            mkdirRecurse(outputDir);
+        }
+    
+        auto lineRange = harFile.byLine(Yes.keepTerminator);
+        if (lineRange.empty)
+        {
+            infoFile.writefln("%s(1): Error: file is empty", harFilename);
+            return No.success;
+        }
+        string delimiter;
+        auto line = lineRange.front;
+        {
+            auto spaceIndex = line.indexOf(' ');
+            if (spaceIndex < 0)
+            {
+                infoFile.writefln("%s(1) Error: first line does not contain a space", harFilename);
+                return No.success;
+            }
+            delimiter = line[0 .. spaceIndex + 1].idup;
+        }
+    
+    LfileLoop:
+        for (;;)
+        {
+            // At this point, line starts with the delimiter
+            auto fileInfoString = line[delimiter.length .. $];
+    
+            // For now this will just be the filename
+            auto fileName = fileInfoString.stripRight;
+            {
+                auto dir = dirName(fileName);
+                if (dir.length > 0)
+                {
+                    auto fullDir = buildPath(outputDir, dir);
+                    if (!extractMkdir(fullDir))
+                        return No.success;
+                }
+            }
+            auto fullFileName = buildPath(outputDir, fileName);
+            infoFile.writefln("creating %s", fullFileName.formatFile);
+            extractedFilenames.put(fullFileName.idup);
+            {
+                auto currentOutputFile = File(fullFileName, "w");
+                scope(exit) currentOutputFile.close();
```

Comment:
Redundant. `File` is refCounted and will be closed once the scope is left, so the behavior is already exactly what you manually defined here.